### PR TITLE
Fix FakeItEasy dependency in nupkg

### PR DESCRIPTION
**Pull request by blairconrad**
We're using rc001 in the build and the package has a dependency on beta002

### CODE REVIEW

**Comment by adamralph** [comment]
Dropping the nuspec munging is the right thing to do. It's only required when you need to pack two or more projects with dependencies between them.

Why are you using FakeItEasy's version number?

File: build.csx [hunk 1]
```diff
@@ -143,3 +133,18 @@ public static void RunDotNet(string workingDirectory, string command, string arg
...
+    {
+        var match = versionMatcher.Match(line);
+        if (match.Success)
+        {
+            return match.Groups[1].Value;
+        }
+    }
+
+    throw new InvalidOperationException($"Can't find FakeItEasy version in {mainProjectPath}");
+}
```
Comment (thomaslevesque) [inline build.csx]: Wouldn't it be easier to parse the JSON?

**Comment by blairconrad** [comment]
> Why are you using FakeItEasy's version number?

It's the FakeItEasy dependency I'm shoving into the nupkg.

File: build.csx (hunk 1 above)
Comment (blairconrad) [reply build.csx]: Maybe!

Comment (blairconrad) [reply build.csx]: You prefer

```c#
public string GetFakeItEasyVersion()
{
    byte[] buffer = File.ReadAllBytes(mainProjectPath);
    XmlReader reader = JsonReaderWriterFactory.CreateJsonReader(buffer, new XmlDictionaryReaderQuotas());

    XElement root = XElement.Load(reader);
    return root.Descendants("FakeItEasy").First().Value;
}
```
?

Comment (thomaslevesque) [reply build.csx]: Wow, I didn't even know it was possible to parse JSON as XML. I don't know if I find it fascinating or horrifying... maybe a bit of both :wink:
I was thinking about using JSON.NET, but maybe it's overkill. If it works this way, it's fine by me. Although I would probably be more specific for finding the dependency (e.g. `root.Element("dependencies").Element("FakeItEasy").Value`)

**Comment by blairconrad** [comment]
@adamralph, you may have been confused because I somehow neglected to upload my nuspec, which had been reverted on disk. Good thing emacs had a copy in memory!

**Comment by blairconrad** [comment]
> Wow, I didn't even know it was possible to parse JSON as XML.

Me neither! In particular, I didn't know that there was a framework-provided way to read JSON.
I've taken your advice and made the path to the version node explicit.

**Comment by adamralph** [comment]
> @adamralph, you may have been confused because I somehow neglected to upload my nuspec, which had been reverted on disk. Good thing emacs had a copy in memory!

👍 that makes sense now. That is actually a very nice way to avoid having to manually maintain the version range. I'm going to steal this idea!

**Comment by blairconrad** [comment]
Thanks, @adamralph. After the version mismatch here, and the one we'd had in FakeItEasy a little while back, I realized that humans can't be trusted.

**Comment by blairconrad** [comment]
Parameterized the method, as seen in [Automatically Sync nupkg and project.json Dependencies](http://blairconrad.com/2017/02/21/automatically-sync-nupkg-and-project.json-dependencies/).
Would y'all prefer if I just kept merging things on my own instead of relying on you for merges? I'm happy either way.

**Comment by thomaslevesque** [comment]
> Would y'all prefer if I just kept merging things on my own instead of relying on you for merges? I'm happy either way.

Me too. It's your project, so I think it makes sense for you to merge yourself when you feel it's ready. We can still do reviews, which you're free to heed or ignore :wink:

**Review by thomaslevesque — APPROVED** [review]

**Comment by blairconrad** [comment]
Okay, @thomaslevesque. Thanks. I'll adjust so I can self-merge. At some point we may want to consider pulling under the FakeItEasy umbrella (unless I misunderstood previous conversations) but until then, I'll act as if it's mine. Also, thanks for the review!

**Comment by blairconrad** [comment]
> until then

or until either of you expresses interest in a more involved role!

**Comment by thomaslevesque** [comment]
> At some point we may want to consider pulling under the FakeItEasy umbrella (unless I misunderstood previous conversations)

👍 

I'm looking at the rest of the code to get more familiar with it.

**Pull request merged**